### ISSUE TITLE
[Bug] Comiple error after Flutter 3.19.0 upgrade

### ISSUE THREAD

**Issue by roman423** [bosskmk/pluto_grid#1016]
### Error
*****/AppData/Local/Pub/Cache/hosted/pub.dev/pluto_grid-7.0.2/lib/src/manager/shortcut/pluto_grid_shortcut.dart(33,39): error GC2F972A8: The argument type 'RawKeyEvent' can't be assigned to the parameter type 'KeyEvent'. [C:\Daten\Workspaces\AndroidStudio\name.deimann.hc\hc_flutter\build\windows\x64\flutter\flutter_assemble.vcxproj]
*****/AppData/Local/Pub/Cache/hosted/pub.dev/pluto_grid-7.0.2/lib/src/manager/shortcut/pluto_grid_shortcut.dart(33,46): error GC2F972A8: The argument type 'RawKeyboard' can't be assigned to the parameter type 'HardwareKeyboard'. [C:\Daten\Workspaces\AndroidStudio\name.deimann.hc\hc_flutter\build\windows\x64\flutter\flutter_assemble.vcxproj]

### Environment
Flutter 3.19.0, on Microsoft Windows [Version 10.0.22631.3155], locale de-DE)
Dart version 3.3.0
PlutoGrid  v7.0.2

**Comment by Yusufihsangorgel** [comment]
+++

**Comment by bigedev** [comment]
+++

**Comment by GayeKone222** [comment]
same problem. any solutions ?

**Comment by Macacoazul01** [comment]
I have a fork of this repo and landed this fix on there.
Not sure if everything is working fine as it should, but the code is running again.

Just use this in your pubspec:
  ```
pluto_grid:
    git: https://github.com/Macacoazul01/pluto_grid
```

**Comment by aymswick** [comment]
_+++ and "me too" comments are noisy and unhelpful, just use the reactions button_

This project is in a confusing state, to see what I mean please read this thread: 
https://github.com/bosskmk/pluto_grid/issues/968 

The tl;dr is:
- Original author/maintainer has abandoned and nobody can write to this repo
- Someone forked it already and has started making changes (though they are essentially 1 person "doing their best") and partially released the functionality of the original package at [pluto_grid_plus](https://github.com/doonfrs/pluto_grid_plus). Notably, the `pluto_grid_export` package has not been republished. I believe while they have not answered issues for a month, they have done prior work to migrate the changes represented by some PRs in *this* repo and they are moving in a positive direction, help is needed! As of yesterday, this particular bug is still present in `pluto_grid_plus`. There is discussion of opening up the fork to more maintainers. 
- There is now a 3rd fork above which contains a replication of the fixes to the `RawKeyEvent` breaking change that someone put up a PR for months ago (though I don't know how anyone would discover this due to its nondescript name [FIX BUG ON WEB] https://github.com/bosskmk/pluto_grid/pull/979

Sorry there isn't an easier answer to your bug discovery, I'm just a guy in the same situation a couple days ahead of you; trying to spread the knowledge so we can all work in a single direction and not create more noise / future abandoned forks.

**Comment by ldsouza93** [comment]
me2

**Comment by kiskelover** [comment]
same problem

flutter 3.19

**Comment by jayakumarzoftsol** [comment]
Same problem . My app broke due to this. urgent help needed :)

**Comment by jayakumarzoftsol** [comment]
> I have a fork of this repo and landed this fix on there. Not sure if everything is working fine as it should, but the code is running again.
> 
> Just use this in your pubspec:
> 
> ```
> pluto_grid:
>   git: https://github.com/Macacoazul01/pluto_grid
> ```

No Its not working showing the dependancy issue with pluto_grid_export

**Comment by Macacoazul01** [comment]
Try the one @aymswick said, pluto_grid_plus

**Comment by pasaneramusugoda** [comment]
> _+++ and "me too" comments are noisy and unhelpful, just use the reactions button_
> 
> This project is in a confusing state, to see what I mean please read this thread: #968
> 
> The tl;dr is:
> 
> * Original author/maintainer has abandoned and nobody can write to this repo
> * Someone forked it already and has started making changes (though they are essentially 1 person "doing their best") and partially released the functionality of the original package at [pluto_grid_plus](https://github.com/doonfrs/pluto_grid_plus). Notably, the `pluto_grid_export` package has not been republished. I believe while they have not answered issues for a month, they have done prior work to migrate the changes represented by some PRs in _this_ repo and they are moving in a positive direction, help is needed! As of yesterday, this particular bug is still present in `pluto_grid_plus`. There is discussion of opening up the fork to more maintainers.
> * There is now a 3rd fork above which contains a replication of the fixes to the `RawKeyEvent` breaking change that someone put up a PR for months ago (though I don't know how anyone would discover this due to its nondescript name [FIX BUG ON WEB] [[FIX BUG ON WEB] #979](https://github.com/bosskmk/pluto_grid/pull/979)
> 
> Sorry there isn't an easier answer to your bug discovery, I'm just a guy in the same situation a couple days ahead of you; trying to spread the knowledge so we can all work in a single direction and not create more noise / future abandoned forks.

https://github.com/doonfrs/pluto_grid_plus/issues/4#issuecomment-1951236317

Seems `pluto_grid_plus_export: ^1.0.4` is now available.

**Comment by hombre-geek** [comment]
I have the same problem as mentioned, the solution was to return to Flutter version 3.16.8 on channel stable. I hope they can solve it soon, so I can continue working with this package.

**Comment by ecoant** [comment]
> I have the same problem as mentioned, the solution was to return to Flutter version 3.16.8 on channel stable. I hope they can solve it soon, so I can continue working with this package.

buddy, nobody is going to solve it on _this_ repository. All the info you need is present in the comments above. Switch to the `pluto_grid_plus` package. The fix is merged in there.

**Comment by dsyrstad** [comment]
> > * Original author/maintainer has abandoned and nobody can write to this repo

This is not true. `pluto_grid` 8.0.0 was released on 2024-03-01 which solves the issue with Flutter 3.19.x.

But for me, the problem is now `pluto_grid_export` is now incompatible with `pluto_grid` 8.0.0:

```
Because pluto_grid_export 1.0.6 depends on pluto_grid >=3.0.2 <8.0.0 and <app> depends on pluto_grid 8.0.0, pluto_grid_export 1.0.6 is forbidden.
```

**Comment by ecoant** [comment]
> > > * Original author/maintainer has abandoned and nobody can write to this repo
> 
> This is not true. `pluto_grid` 8.0.0 was released on 2024-03-01 which solves the issue with Flutter 3.19.x.
> 
> But for me, the problem is now `pluto_grid_export` is now incompatible with `pluto_grid` 8.0.0:
> 
> ```
> Because pluto_grid_export 1.0.6 depends on pluto_grid >=3.0.2 <8.0.0 and <app> depends on pluto_grid 8.0.0, pluto_grid_export 1.0.6 is forbidden.
> ```

It absolutely was true at the time of my comment. It looks like the maintainer of this repo has returned to partially fix that error, but as you point out, you still won't get anywhere with `export`. I'm not sure what's going on here but you have an alternative option in `pluto_grid_plus`.
